components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/ha/HAEventListener.java

```diff
@@ -61,7 +61,8 @@ public void memberAdded(NodeDetail nodeDetail) {
             for (SourceHandler sourceHandler : registeredSourceHandlers.values()) {
                 ((HACoordinationSourceHandler) sourceHandler).setPassiveNodeAdded(true);
             }
-            if (!nodeDetail.isCoordinator()) {
+            if (!nodeDetail.getNodeId().equals(clusterCoordinator.getLeaderNode().getNodeId()) && nodeDetail
+                    .getPropertiesMap() != null) {
```

Comment:
@NisalaNiroshana do we need this since its mandatory in deloyment yaml ?

Reply:
@sajithshn @dnwick When the passive node joins the cluster initially, it will contain the properties map. But there are some scenarios where the passive node is removed from the cluster and come again to the cluster (Without restarting the server). In that scenarioes, passive does not have the properties map. But we keep the properties map which is retrieved at the initial memeber join in the active node. So we use that properties map to send data.